File: DiscoJava/DiscoVirtual/src/main/java/org/example/ItemArmazenamento.java
package org.example;

import java.util.List;

public interface ItemArmazenamento {

    public void apagar();

    public long getTamanhoKB();

    public Boolean add(ItemArmazenamento item);

    public Boolean remove(ItemArmazenamento item);

    public List<ItemArmazenamento> getItens();

    public String getTipo();

}

File: DiscoTs/src/entities/ItemArmazenamento.ts
interface ItemArmazenamento {
  apagar(): Promise<void>;

  getTamanhoKB(): Promise<number>;

  add(item: ItemArmazenamento): Promise<void>;

  remove(item: ItemArmazenamento): Promise<void>;

  getItens(): Promise<ItemArmazenamento[]>;

  getTipo(): Promise<string>;
}

export { ItemArmazenamento };
